In [1]:
HTML(read(open("style.html"), String))

HTML{String}("<link href='https://fonts.googleapis.com/css?family=Alegreya+Sans:100,300,400,500,700,800,900,100italic,300italic,400italic,500italic,700italic,800italic,900italic' rel='stylesheet' type='text/css'>\r\n<link href='https://fonts.googleapis.com/css?family=Arvo:400,700,400italic' rel='stylesheet' type='text/css'>\r\n<link href='https://fonts.googleapis.com/css?family=PT+Mono' rel='stylesheet' type='text/css'>\r\n<link href='https://fonts.googleapis.com/css?family=Shadows+Into+Light' rel='stylesheet' type='text/css'>\r\n<link href='https://fonts.googleapis.com/css?family=Philosopher:400,700,400italic,700italic' rel='stylesheet' type='text/css'>\r\n\r\n<style>\r\n\r\n@font-face {\r\n    font-family: \"Computer Modern\";\r\n    src: url('https://mirrors.ctan.org/fonts/cm-unicode/fonts/otf/cmunss.otf');\r\n}\r\n\r\n.container { width: 100% }\r\n\r\n/* Formatting for header cells */\r\n.text_cell_render h1 {\r\n    font-family: 'Philosopher', sans-serif;\r\n    font-weight: 400;\r\n    font-size: 2.2em;\r\n    line-height: 100%;\r\n    color: rgb(0, 80, 120);\r\n    margin-bottom: 0.1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\t\r\n.text_cell_render h2 {\r\n    font-family: 'Philosopher', serif;\r\n    font-weight: 400;\r\n    font-size: 1.9em;\r\n    line-height: 100%;\r\n    color: rgb(200,100,0);\r\n    margin-bottom: 0.1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\t\r\n\r\n.text_cell_render h3 {\r\n    font-family: 'Philosopher', serif;\r\n    margin-top:12px;\r\n    margin-bottom: 3px;\r\n    font-style: italic;\r\n    color: rgb(94,127,192);\r\n}\r\n\r\n.text_cell_render h4 {\r\n    font-family: 'Philosopher', serif;\r\n}\r\n\r\n.text_cell_render h5 {\r\n    font-family: 'Alegreya Sans', sans-serif;\r\n    font-weight: 300;\r\n    font-size: 16pt;\r\n    color: grey;\r\n    font-style: italic;\r\n    margin-bottom: .1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\r\n\r\n.text_cell_render h6 {\r\n    font-family: 'PT Mono', sans-serif;\r\n    font-weight: 300;\r\n    font-size: 10pt;\r\n    color: grey;\r\n    margin-bottom: 1px;\r\n    margin-top: 1px;\r\n}\r\n\r\n.text_cell_render em {\r\n    font-family: 'Philosopher', sans-serif;\r\n    color:        blue;\r\n    background-color: rgb(255,220,180);\r\n    font-size:    110%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   100;\r\n}\r\n\r\n.text_cell_render b {\r\n    color:            rgb(255,195,195);\r\n    background-color: rgb(0,0,0);\r\n    font-size:    110%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   650;\r\n}\r\n\r\n.text_cell_render tt {\r\n    font-size:    120%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   150;\r\n}\r\n\r\n.Codemirror {\r\n    font-family: \"PT Mono\", sans-serif;\r\n    font-size: 100%;\r\n}\r\n\r\n</style>")

## Table of Contents

It is recommended to read the notebooks in the order they are listed below. You may skip any test notebooks.

### Basic Information & Introduction
 - [1.0 - Julia Programming Language.ipynb](<1.0%20-%20Julia%20Programming%20Language.ipynb>)
 - [1.1 - Chess.jl Library.ipynb](<1.1%20-%20Chess.jl%20Library.ipynb>)
 - [2.0 - Gameplay.ipynb](<2.0%20-%20Gameplay.ipynb>)
 - [2.1 - Board.ipynb](<2.1%20-%20Board.ipynb>)
 - [2.1.T - Board (Test).ipynb](<2.1%20-%20Board%20%28Test%29.ipynb>)


### Random Movement Baseline
 - [2.2 - Random AI.ipynb](<2.2%20-%20Random%20AI.ipynb>)


### Minimax Algorithm & Position Evaluation
 - [3.0 - Piece-Values.ipynb](<3.0%20-%20Piece-Values.ipynb>)
 - [3.0.T - Piece-Values (Test).ipynb](<3.0.T%20-%20Piece-Values%20%28Test%29.ipynb>)
 - [3.1 - Evaluation.ipynb](<3.1%20-%20Evaluation.ipynb>)
 - [3.1.T - Evaluation (Test).ipynb](<3.1.T%20-%20Evaluation%20%28Test%29.ipynb>)
 - [3.2 - Minimax.ipynb](<3.2%20-%20Minimax.ipynb>)
 - [3.2.T - Minimax (Test).ipynb](<3.2.T%20-%20Minimax%20%28Test%29.ipynb>)
 - [3.3 - Minimax AI.ipynb](<3.3%20-%20Minimax%20AI.ipynb>)


### Memoization Optimization
 - [4.0 - Memoisierung - Zobrist-Hashing.ipynb](<4.0%20-%20Memoisierung%20-%20Zobrist-Hashing.ipynb>)
 - [4.0.T - Memoisierung - Zobrist-Hashing (Test).ipynb](<4.0.T%20-%20Memoisierung%20-%20Zobrist-Hashing%20%28Test%29.ipynb>)
 - [4.1 - Memoisierung - Minimax.ipynb](<4.1%20-%20Memoisierung%20-%20Minimax.ipynb>)
 - [4.1.T - Memoisierung - Minimax.ipynb](<4.1.T%20-%20Memoisierung%20-%20Minimax%20%28Test%29.ipynb>)
 - [4.2 - Memoisierung - Minimax AI.ipynb](<4.1%20-%20Memoisierung%20-%20Minimax%20AI.ipynb>)


### Alpha-Beta Pruning Optimization
 - [5.0 - Alpha-Beta-Pruning.ipynb](<5.0%20-%20Alpha-Beta-Pruning.ipynb>)
 - [5.0.T - Alpha-Beta-Pruning (Test).ipynb](<5.0.T%20-%20Alpha-Beta-Pruning%20%28Test%29.ipynb>)
 - [5.1 - Alpha-Beta-Pruning AI.ipynb](<5.1%20-%20Alpha-Beta-Pruning%20AI.ipynb>)
 
 
### Iterative Deepening
 - [6.0 - Iterative Deepening.ipynb](<6.0%20-%20Iterative%20Deepening.ipynb>)
 - [6.0.T - Iterative Deepening (Test).ipynb](<6.0.T%20-%20Iterative%20Deepening%20%28Test%29.ipynb>)
